nemo/src/parser/ast/token.rs

```diff
@@ -316,7 +316,7 @@ impl<'a> Token<'a> {
             ParserContext::token(TokenKind::Name),
             recognize(pair(
                 alpha1,
-                many0(alt((alphanumeric1, tag("_"), tag("-"), tag("%")))),
+                many0(alt((alphanumeric1, tag("_"), tag("%")))),
```

Comment:
As far as I can tell, this breaks compatibility with SPARQL, since prefix names may contain `-`: https://www.w3.org/TR/sparql11-query/#rPN_CHARS

Reply:
Hm, but its hard to fix without running into the problem above. Consider the following program:

```
@prefix my-prefix <https://prefix.com> .

result(my-prefix:test) :- body(?x).
```
Now `my-prefix:test` can either be the constant with the prefix `my-prefix` or the arithmetic operation `my <minus> prefix:test`. 

We talked about fixing this sort of ambiguity in the context of having `:` as key-value separator in maps by having the parser state include all known prefixes,. But this would require a more thorough extension of the parser that is not in the scope of this quick bug-fix.

Reply:
I will create an issue about this so to keep track of this